backend/api/professor/grading_materials.py

```
@@ -0,0 +1,333 @@
+            )
+        
+        # Validate file
+        allowed_extensions = {'.pdf', '.doc', '.docx', '.txt', '.md'}
+        
+        # Validate file type
+        file_extension = os.path.splitext(file.filename)[1].lower()
+        if file_extension not in allowed_extensions:
+            raise HTTPException(
+                status_code=400,
+                detail=f"File type {file_extension} not supported. Allowed: {', '.join(allowed_extensions)}"
+            )
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)